heaps/min_heap.py

```diff
@@ -21,18 +21,32 @@ def __init__(self):
     def add(self, key, value = None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(log n)
+            Space Complexity: O(log n)
         """
-        pass
+        if not value:
+            new_heapnode = HeapNode(key, key)
+        else:
+            new_heapnode = HeapNode(key, value)
+        self.store.append(new_heapnode)
+        index = len(self.store) - 1
+        self.heap_up(index)
+        return None
 
     def remove(self):
         """ This method removes and returns an element from the heap
             maintaining the heap structure
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(log n)
+            Space Complexity: O(1)
```

Comment:
👍 However the space complexity is O(log n) due to the recursive `heap_down` method.